packages/framework/src/services/smb/smb-broker.client.ts

```diff
@@ -0,0 +1,47 @@
+import { BrokerClient, BrokerClientFactory, IBrokerClientOptions } from '@pequehq/smb-client';
+
+import { BrokerAddressInvalidError } from '../../models';
+import { BrokerProxy } from '../../models/interfaces/broker-proxy.interface';
+
+export interface SmbPublishPayload {
+  topic: string;
+  data: string;
+}
+
+export interface SmbSubscribePayload {
+  topic: string;
+  data: string;
+}
+
+export class SmbBrokerClient extends BrokerProxy<unknown> {
+  #client: BrokerClient;
+
+  async connect(): Promise<void> {
+    const calculateHostPort = (): IBrokerClientOptions => {
+      const split = this.broker.split(':');
+
+      if (!split[1] || isNaN(Number(split[1]))) {
+        throw new BrokerAddressInvalidError(this.broker);
+      }
+
+      return {
+        host: split[0],
+        port: Number(split[1]),
+      };
```

Comment:
Here we can take advantage of destructuring assignment to have more meaningful names
```suggestion
      const [host, port] = this.broker.split(':');

      if (!port || isNaN(Number(port))) {
        throw new BrokerAddressInvalidError(this.broker);
      }

      return {
        host,
        port: Number(port),
      };
```